_spinnaker_install_admin_guides/install-on-k8s.md

```diff
@@ -0,0 +1,710 @@
+
+From the `workstation machine` where `kubectl` is installed, install the NGINX ingress controller components:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/mandatory.yaml
+```
+
+Then, install the NGINX ingress controller AWS-specific service:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/aws/service-l4.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/aws/patch-configmap-l4.yaml
+```
+
+### Set up an Ingress for `spin-deck` and `spin-gate`
+
+Get the external IP for the NGINX ingress controller:
+
+```
+kubectl get svc -n ingress-nginx
+```
+
+The command returns a DNS name or IP address in the `EXTERNAL-IP` field.
+
+If you stood up a new NGINX ingress controller, you can likely use this value (IP address or DNS name) for your ingress.
+
+For example, if the command returns `abcd1234abcd1234abcd1234abcd1234-123456789.us-west-2.elb.amazonaws.com`, then you can use `abcd1234abcd1234abcd1234abcd1234-123456789.us-west-2.elb.amazonaws.com` for the `SPINNAKER_ENDPOINT` in the following steps. If the command returns `55.55.55.55`, then use `55.55.55.55` for the `SPINNAKER_ENDPOINT`.
+
+If you use an existing NGINX ingress controller or other services are likely to be using the same NGINX ingress controller, create a DNS entry that points at the NGINX ingress controller endpoint you are using for Spinnaker. You can create either a `CNAME Record` that points at the DNS name or an `A Record` that points at the IP address.
+
+For the example `abcd1234abcd1234abcd1234abcd1234-123456789.us-west-2.elb.amazonaws.com` DNS name, do the following:
+* Create a CNAME pointing `spinnaker.domain.com` at `abcd1234abcd1234abcd1234abcd1234-123456789.us-west-2.elb.amazonaws.com`
+* Put `spinnaker.domain.com` in the `host` field in the below manifest and uncomment it
+* Use `spinnaker.domain.com` for the `SPINNAKER_ENDPOINT` in the below steps
+* (Alternately, for testing, create an `/etc/hosts` entry pointing `spinnaker.domain.com` at the IP address that `abcd1234abcd1234abcd1234abcd1234-123456789.us-west-2.elb.amazonaws.com` resolves to)
+
+For the `55.55.55.55` IP address example, do the following:
+* Create an `A Record` pointing to `spinnaker.domain.com` at `55.55.55.55`
+* Put `spinnaker.domain.com` in the `host` field in the below manifest and uncomment it
+* Use `spinnaker.domain.com` for my SPINNAKER_ENDPOINT in the below steps
+* (Alternately, for testing, create an `/etc/hosts` entry pointing `spinnaker.domain.com` at `55.55.55.55`)
+
+Create a Kubernetes Ingress manifest to expose `spin-deck` and `spin-gate`.
+
+Create a file called `spin-ingress.yml` with the following content.
+
+(Make sure the hosts and namespace match your actual host and namespace.)
+
+```bash
+---
+apiVersion: networking.k8s.io/v1
```

Comment:
I had to use: 
apiVersion: extensions/v1beta1
otherwise, I got this error: 
error: unable to recognize "spin-ingress.yml": no matches for kind "Ingress" in version "networking.k8s.io/v1"

Reply:
Oh crap.  Okay so networking.k8s.io/v1 was introduced in 1.14, and extensions/v1beta1 is deprecated as of 1.20.  So yeah we should probably (currently) go with the older version.